src/libutil/ansicolor.hh

```diff
@@ -2,16 +2,19 @@
 
 namespace nix {
 
+// A forward declaration from `util.hh`
+bool shouldANSI();
+
 /* Some ANSI escape sequences. */
-#define ANSI_NORMAL "\e[0m"
-#define ANSI_BOLD "\e[1m"
-#define ANSI_FAINT "\e[2m"
-#define ANSI_ITALIC "\e[3m"
-#define ANSI_RED "\e[31;1m"
-#define ANSI_GREEN "\e[32;1m"
-#define ANSI_YELLOW "\e[33;1m"
-#define ANSI_BLUE "\e[34;1m"
-#define ANSI_MAGENTA "\e[35;1m"
-#define ANSI_CYAN "\e[36;1m"
+#define ANSI_NORMAL  "\e[0m"
+#define ANSI_BOLD    "\e[1m"
+#define ANSI_FAINT   "\e[2m"
+#define ANSI_ITALIC  "\e[3m"
+#define ANSI_RED     (::std::string( ::nix::shouldANSI() ? "\e[31;1m" : ""))
+#define ANSI_GREEN   (::std::string( ::nix::shouldANSI() ? "\e[32;1m" : ""))
+#define ANSI_YELLOW  (::std::string( ::nix::shouldANSI() ? "\e[33;1m" : ""))
+#define ANSI_BLUE    (::std::string( ::nix::shouldANSI() ? "\e[34;1m" : ""))
+#define ANSI_MAGENTA (::std::string( ::nix::shouldANSI() ? "\e[35;1m" : ""))
+#define ANSI_CYAN    (::std::string( ::nix::shouldANSI() ? "\e[36;1m" : ""))
```

Comment:
I don't think this should be dynamic. The current approach is to use ANSI color unconditionally when constructing messages, and then strip ANSI codes when printing to a non-terminal. That way, messages from the daemon (whose stderr isn't connected to a terminal) get colorized properly.

Reply:
In that case, I am closing the PR since I dont know how to implement it.